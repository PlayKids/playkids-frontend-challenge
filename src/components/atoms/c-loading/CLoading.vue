<template>
  <div class="background">
    <spinner class="spinner" />
  </div>
</template>

<script>
import spinner from '@/assets/img/spinner.svg';

export default {
  name: 'CLoading',

  components: {
    spinner
  }
};
</script>

<style scoped>
.background {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
}

.spinner {
  height: 100%;
}
</style>
