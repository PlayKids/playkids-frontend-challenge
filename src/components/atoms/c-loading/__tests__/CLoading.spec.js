import { mount } from '@vue/test-utils';
import CLoading from '../CLoading.vue';

describe('CLoading component', () => {
  it('renders correctly', () => {
    const wrapper = mount(CLoading, {
      stubs: ['spinner']
    });
    expect(wrapper).toMatchSnapshot();
  });
});
