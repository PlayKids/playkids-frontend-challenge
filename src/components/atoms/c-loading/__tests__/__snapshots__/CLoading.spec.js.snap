// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CLoading component renders correctly 1`] = `
<div class="spinner-background">
  <spinner-stub class="spinner"></spinner-stub>
</div>
`;
